tests/unit/s2n_connection_preferred_keyshare_test.c

```diff
@@ -0,0 +1,128 @@
+    struct s2n_config *config = NULL;
+
+    /* Test error case for setting preferred keyshares prior to ecc_preferences being configured */
+    {
+        EXPECT_FAILURE(s2n_connection_set_keyshare_by_group_for_testing(conn, TLS_EC_CURVE_SECP_256_R1));
+    }
+    /* Test sending empty keyshare */
+    {
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_keyshare_by_group_for_testing(conn, 0));
+        /* lsb is set */
+        EXPECT_TRUE(conn->preferred_key_shares & 1);
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+    /* Test sending single keyshares: p-256 */
+    {
+        EXPECT_NOT_NULL(config = s2n_config_new());
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+        /* Explicity set default security_policy */
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "20170210"));
+        EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+        const struct s2n_ecc_preferences *ecc_pref = NULL;
+        EXPECT_SUCCESS(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
+        EXPECT_NOT_NULL(ecc_pref);
+        /* Test success for setting preferred keyshares bitmap for curve p-256 after ecc_preferences is configured
+         * Default security policy has ecc_preferences in the following order: p-256, p-384 */
+        EXPECT_SUCCESS(s2n_connection_set_keyshare_by_group_for_testing(conn, TLS_EC_CURVE_SECP_256_R1));
+        /* lsb is not set */
+        EXPECT_FALSE(conn->preferred_key_shares & 1);
+        /* Bitmap value for p-256 set is 00000010 */
+        EXPECT_TRUE((conn->preferred_key_shares >> 1) & 1);
+        EXPECT_SUCCESS(s2n_config_free(config));
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+    /* Test success for setting multiple preferred keyshares: x25519 and p-384 */
+    if (s2n_is_evp_apis_supported())
+    {
+        EXPECT_SUCCESS(s2n_enable_tls13());
+        EXPECT_NOT_NULL(config = s2n_config_new());
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+        /* Explicity set default_tls13 security_policy */
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "20190801"));
+        EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+        /* Default security policy for TLS1.3 has ecc_preferences in the following order: x25519, p-256, p-384.*/
+        EXPECT_SUCCESS(s2n_connection_set_keyshare_by_group_for_testing(conn, TLS_EC_CURVE_ECDH_X25519));
+        /* lsb is not set */
+        EXPECT_FALSE(conn->preferred_key_shares & 1);
+        /* Bitmap value for x25519 set is 00000010 */
+        EXPECT_TRUE((conn->preferred_key_shares >> 1) & 1);
+        EXPECT_SUCCESS(s2n_connection_set_keyshare_by_group_for_testing(conn, TLS_EC_CURVE_SECP_384_R1));
+        /* Bitmap value for p-384 and x25519 set is 00001010 */
+        EXPECT_FALSE((conn->preferred_key_shares >> 2) & 1);
+        EXPECT_TRUE((conn->preferred_key_shares >> 3) & 1);
```

Comment:
All these bit operations make these tests hard to follow / verify. Can you either use the macros you define in response to the other comments on this PR, or define test-specific macros to make the tests more readable?